### PR TITLE
fix(pie-textarea): DSW-2470 allow resize on programmatic value change

### PR DESCRIPTION
**Pull request by jamieomaguire**
## Describe your changes (can list changeset entries if preferable)
- Ensures that when the `value` property is updated programmatically (e.g., through a Storybook control), the text area resizes correctly, just as it would if a user were typing directly into the input.

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1 - Fernando
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 - @kevinrodrigues 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Review by fernandofranca — DISMISSED** [review]

File: packages/components/pie-textarea/src/index.ts [hunk 1]
```diff
@@ -158,8 +157,13 @@ export class PieTextarea extends FormControlMixin(RtlMixin(LitElement)) implemen
      * Handles data processing in response to the input event. The native input event is left to bubble up.
      * @param event - The input event.
```
[the remark this answers is not in the document]
Comment (jamieomaguire) [reply packages/components/pie-textarea/src/index.ts:158]: done @kevinrodrigues cheers 👍

File: packages/components/pie-textarea/test/visual/pie-textarea.spec.ts [hunk 2]
```diff
@@ -159,6 +159,23 @@ test.describe('Resize mode:', () => {
...
+                    resize: 'auto',
+                } as PieTextarea,
+            });
+
+            await page.evaluate(() => {
+                const textarea = document.querySelector('pie-textarea') as PieTextarea;
+                textarea.value = 'The default height is enough for two lines of text, but it should grow if you type more.';
+            });
+
+            await page.waitForTimeout(250); // Wait for throttled resize event to fire.
```
[the remark this answers is not in the document]
Comment (jamieomaguire) [reply packages/components/pie-textarea/test/visual/pie-textarea.spec.ts:174]: Good question, I am not actually sure. Will look into it now!

**Review by kevinrodrigues — APPROVED** [review]

File: packages/components/pie-textarea/test/visual/pie-textarea.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamieomaguire) [reply packages/components/pie-textarea/test/visual/pie-textarea.spec.ts:174]: I will refactor this to constant in a follow up PR 🤝 It's basically just a small timeout that should ensure consistency when awaiting for browser events/renders to trigger

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**